lib/ecto/migration.ex

```diff
@@ -783,6 +783,37 @@ defmodule Ecto.Migration do
     Runner.subcommand {:add, column, type, opts}
   end
 
+  @doc """
+  Adds a column if it not exists yet when altering a table.
+
+  If the `type` value is a `%Reference{}`, it is used to remove the constraint.
+
+  `type` and `opts` are exactly the same as in `add/3`, and
+  they are used when the command is reversed.
+
+  ## Examples
+
+      alter table("posts") do
+        add_if_not_exists :title, :string, default: ""
+      end
+
+  """
+  def add_if_not_exists(column, type, opts \\ [])
+
+  def add_if_not_exists(column, :datetime, _opts) when is_atom(column) do
+    raise ArgumentError, "the :datetime type in migrations is not supported, " <>
+                         "please use :utc_datetime or :naive_datetime instead"
+  end
```

Comment:
We should probably remove these clauses and do the validation inside `validate_type!` but that is the job of another PR. :D